eo-runtime/src/main/java/EOorg/EOeolang/EOseq.java

```diff
@@ -58,25 +59,33 @@ public EOseq(final Phi sigma) {
     @Override
     public Phi lambda() {
         final Phi steps = this.take("steps");
-        final Long length = new Dataized(
-            steps.take("length")
-        ).take(Long.class);
-        for (long idx = 0; idx < length - 1; ++idx) {
-            new Dataized(
-                new PhWith(
-                    steps.take("at").copy(),
-                    0, new Data.ToPhi(idx)
-                )
-            ).take();
+        final Phi[] items = toArray(steps);
+        for (int ind = 0; ind < items.length - 1; ++ind) {
+            new Dataized(items[ind]).take();
         }
         final Phi ret;
-        if (length > 0) {
+        if (items.length > 0) {
             final Phi last = steps.take("at").copy();
-            last.put(0, new Data.ToPhi(length - 1));
+            last.put(0, new Data.ToPhi((long) items.length - 1L));
             ret = last;
         } else {
             ret = new Data.ToPhi(false);
         }
         return ret;
     }
+
+    private static Phi[] toArray(final Phi args) {
+        final int length = Math.toIntExact(
+            new Dataized(
+                args.take("length")
+            ).take(Long.class)
```

Comment:
@levBagryansky to cast to `long` it's better to use `.intValue()`

Reply:
@maxonfjvipon `Dataized` does not have `intValue` method

Reply:
@levBagryansky `Long` has

Reply:
@maxonfjvipon I see. The difference is that `toIntExact` throws exception in case of overflow. So it is safer.. What if mistake happened and `new args.take("length")).take(Long.class)` returned too big number.

Reply:
@levBagryansky ok, let it be so